lib/matplotlib/backends/backend_webagg_core.py

```diff
@@ -189,18 +193,14 @@ def get_diff_image(self):
                 output = buff
             else:
                 self.set_image_mode('diff')
-                last_buffer = (np.frombuffer(self._last_renderer.buffer_rgba(),
-                                             dtype=np.uint32)
-                               .reshape((renderer.height, renderer.width)))
-                diff = buff != last_buffer
+                diff = buff != self._last_buff
                 output = np.where(diff, buff, 0)
 
             buf = BytesIO()
```

Comment:
I notice that there are both `buff` and `buf` in this code (I realise you didn't introduce or touch the last one). Do you think you could attempt to clear up what the difference between them is, and perhaps make the variable names self explanatory?

Reply:
I haven't left a comment because I'm not 100% sure this is the reason. But here's the idea of what I might leave:

We differentiate between `buff` and `buf` because we cannot directly return `buff` as that is a reference to a location in memory that may change later.

There may also be something happening with the save where the buffer gets converted from whatever the renderer uses into a png buffer.

Reply:
See #19117.